samples/corda/fungible-house-token/extension-flows/AssetTransferHelperFlows.kt

```diff
@@ -0,0 +1,246 @@
+    val pledgeId: String,
+    val expiryTimeSecs: String
+) : FlowLogic<ByteArray>() {
+    /**
+     * The call() method captures the logic to fetch [AssetClaimStatusState] vault state from importing n/w.
+     *
+     * @return Returns ByteArray.
+     */
+
+    @Suspendable
+    override fun call(): ByteArray {
+
+        println("Inside GetAssetClaimStatusByPledgeId(), pledgeId: $pledgeId and expiryTimeSecs: $expiryTimeSecs.")
+
+        println("Creating empty fungible token asset JSON.")
+        var marshalledBlankAssetJson = subFlow(MarshalFungibleToken("dummy-token-type", 0L, "dummy-owner-cert"))
+
+        return subFlow(GetAssetClaimStatusState(pledgeId, expiryTimeSecs, marshalledBlankAssetJson))
+    }
+}
+
+/**
+ * The GetHouseTokenJsonStringFromStatePointer flow fetches the fungible token state from its state pointer [AssetPledgeState].assetStatePointer
+ * and creates marshalled JSON encoded object which is returned.
+ * This function is called by the exporting network in the context of interop-query from the importing network
+ * to the exporting network before performing claim on remote asset.
+ *
+ * @property assetPledgeState The (interop) vault state that represents the pledge details on the ledger.
+ */
+@InitiatingFlow
+@StartableByRPC
+class GetHouseTokenJsonStringFromStatePointer(
+    val assetPledgeState: AssetPledgeState
+) : FlowLogic<String>() {
+    @Suspendable
+    override fun call(): String {
+
+        if (assetPledgeState.assetStatePointer == null) {
+            // Typically, [AssetPledgeState].assetStatePointer will be null only in the case of pledge details not
+            // being available for a given pledgeId. The flow GetAssetPledgeStatus in AssetTransferFlows sets this
+            // pointer to null if the pledge-state is not available in the context of the interop-query from the
+            // importing n/w to the exporting n/w. Hence return empty string, and this will not be passed to the
+            // JSON unmarshalling method GetTokenStateAndContractId since the expiryTime will be elapsed for the
+            // claim to happen (i.e., if assetStatePointer is null, then expiryTimeSecs will be set to past time).
+            return ""
+        }
+
+        val assetStatePointer: StaticPointer<ContractState> = assetPledgeState.assetStatePointer!!
+        val assetState = assetStatePointer.resolve(serviceHub).state.data as FungibleToken
+
+        @Suppress("UNCHECKED_CAST")
+        val fungibleHouseTokenPointer: TokenPointer<FungibleHouseTokenState> = assetState.amount.token.tokenType as TokenPointer<FungibleHouseTokenState>
+        val fungibleHouseTokenState = fungibleHouseTokenPointer.pointer.resolve(serviceHub).state.data
+
+        println("Creating fungible token asset JSON from StatePointer.")
+        var marshalledAssetJson =
+            subFlow(MarshalFungibleToken(fungibleHouseTokenState.symbol, assetState.amount.quantity, assetPledgeState.lockerCert))
```

Comment:
same comment as in SimpleAssetFlows file, move contents of this subflow to here.

Or you can declare functions independently and call it, instead of calling it as subFlows, because subflows are expensive. (check CommonHelperFunctions.kt in corda-interop-app for reference).